lib/index.js

```diff
@@ -21,11 +32,31 @@ class WriteJsonWebpackPlugin {
   }
 
   createObj(compilation, callback) {
-    const json = JSON.stringify(
-      this.options.object,
-      null,
-      this.options.pretty ? 2 : null
-    );
+    let obj = this.options.object;
+
+    if (this.options.extendFile) {
+      if (Array.isArray(compilation.fileDependencies)) {
+        if (
+          compilation.fileDependencies.indexOf(this.options.extendFile) === -1
+        ) {
+          compilation.fileDependencies.push(this.options.extendFile);
+        }
+      } else if (!compilation.fileDependencies.has(this.options.extendFile)) {
+        compilation.fileDependencies.add(this.options.extendFile);
+      }
+
+      try {
+        obj = Object.assign(
+          JSON.parse(fs.readFileSync(this.options.extendFile, "utf8")),
+          this.options.object
```

Comment:
The parameters order is important here to handle the case when both, the passed object and the extended file define the same key. Can you add a spec to ensure `options.object` overwrites `options.extendFile`?

Reply:
https://github.com/aponyx/write-json-webpack-plugin/pull/5/commits/3b54842b7521d9e9d3e212b48255c9e01ce76d3f should be good